tests/test_wave_07.py

```diff
@@ -0,0 +1,237 @@
+import pytest
+from swap_meet.item import Item
+from swap_meet.vendor import Vendor
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+def test_get_by_age_default_age():
+    cloth = Clothing()
+    assert cloth.age == None
+
+def test_get_by_age():
+    cloth = Clothing(condition = 2, age=1)
+    assert cloth.age == 1
+
+def test_get_newest_item():
+    item_a = Item(age=2)
+    item_b = Item(age=3)
+    item_c = Item(age=4)
+    michelle = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+    newest_item = michelle.get_newest_item()
+
+    assert newest_item == item_a
+    assert newest_item.age == pytest.approx(2.0)
+
+
+# def test_get_newest_item_no_matches_is_none():
+#     item_a = Decor(age=10.0)
+#     item_b = Decor(age=10.0)
+#     item_c = Decor(age=4.0)
+#     michelle = Vendor(
+#         inventory=[item_a, item_b, item_c]
+#     )
+
+#     best_item = michelle.get_newest_item("Electronics")
+
+#     assert best_item is None
+
+
+# def test_get_newest_item_with_duplicates():
+#     item_a = Clothing(age=2.0)
+#     item_b = Clothing(age=4.0)
+#     item_c = Clothing(age=4.0)
+#     michelle = Vendor(
+#         inventory=[item_a, item_b, item_c]
+#     )
+
+#     best_item = michelle.get_newest_item("Clothing")
+
+#     assert best_item.category == "Clothing"
+#     assert best_item.age == pytest.approx(2.0)
+
+
+# def test_swap_by_newest():
+#     item_a = Decor(age=2.0)
+#     item_b = Electronics(age=4.0)
+#     item_c = Decor(age=4.0)
+#     michelle = Vendor(
+#         inventory=[item_a, item_b, item_c]
+#     )
+
+#     item_d = Clothing(age=2.0)
+#     item_e = Decor(age=4.0)
+#     item_f = Clothing(age=4.0)
+#     ivette = Vendor(
+#         inventory=[item_d, item_e, item_f]
+#     )
+
+#     result = michelle.swap_by_newest(
+#         other=ivette,
+#         my_item="Clothing",
+#         their_item="Decor"
```

Comment:
Were you thinking of defining `swap_by_newest` as swapping the newest of one category with the newest of another category? That's a reasonable interpretation of `swap_by_newest`, but in that case, spend some time making sure the parameters convey that's what your intent is, and make sure that the helper functions you're using internally have  the functionality needed to carry out your objective. Again, thinking through the steps of the feature you're working on to make sure you can clearly describe the operations to yourself can help give us a clear picture of what we eventually need to write as code.